src/WebJobs.Script/Description/CSharp/CSharpFunctionDescriptionProvider.cs

```diff
@@ -101,50 +102,59 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
-                    if (parameterType.IsByRef)
+                    // Is it the trigger parameter?
+                    if (string.Compare(parameter.Name, triggerMetadata.Name, StringComparison.Ordinal) == 0)
                     {
-                        parameterType = parameterType.GetElementType();
+                        descriptors.Add(CreateTriggerParameterDescriptor(parameter, triggerMetadata, triggerType, methodAttributes, triggerNameSpecified));
                     }
+                    else
+                    {
+                        Type parameterType = parameter.ParameterType;
+                        if (parameterType.IsByRef)
+                        {
+                            parameterType = parameterType.GetElementType();
+                        }
 
-                    var descriptor = new ParameterDescriptor(parameter.Name, parameter.ParameterType);
+                        var descriptor = new ParameterDescriptor(parameter.Name, parameter.ParameterType);
 
-                    var binding = bindings.FirstOrDefault(b => string.Compare(b.Name, parameter.Name, StringComparison.Ordinal) == 0);
-                    if (binding != null)
-                    {
-                        CustomAttributeBuilder customAttribute = binding.GetCustomAttribute();
-                        if (customAttribute != null)
+                        var binding = bindings.FirstOrDefault(b => string.Compare(b.Name, parameter.Name, StringComparison.Ordinal) == 0);
+                        if (binding != null)
                         {
-                            descriptor.CustomAttributes.Add(customAttribute);
+                            CustomAttributeBuilder customAttribute = binding.GetCustomAttribute();
+                            if (customAttribute != null)
+                            {
+                                descriptor.CustomAttributes.Add(customAttribute);
+                            }
                         }
-                    }
 
-                    if (parameter.IsOut)
-                    {
-                        descriptor.Attributes |= ParameterAttributes.Out;
-                    }
+                        if (parameter.IsOut)
+                        {
+                            descriptor.Attributes |= ParameterAttributes.Out;
+                        }
 
-                    descriptors.Add(descriptor);
+                        descriptors.Add(descriptor);
+                    }
                 }
-            }
 
-            return descriptors;
+                return descriptors;
+            }
+            catch (CompilationErrorException)
+            {
+                // We were unable to compile the function to get its signature,
+                // setup the descriptor with the default parameters
+                return base.GetFunctionParameters(functionInvoker, functionMetadata, triggerMetadata, methodAttributes, inputBindings, outputBindings);
```

Comment:
This will behave as it did before, setting up the invoker so that it can monitor the script files and if/when executed, the compilation error details are logged, as they were before. This still uses the C# descriptor, just with the same set of parameters used by the other functions.